### ISSUE TITLE
[Bug]: Etcd are storing big key-value pairs, causing all etcd requests being slow.

### ISSUE THREAD

**Issue by soothing-rain** [milvus-io/milvus#16047]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: latest master
- Deployment mode(standalone or cluster): standalone
- SDK version(e.g. pymilvus v2.0.0rc2): N/A
- OS(Ubuntu or CentOS): MacOS
- CPU/Memory: N/A
- GPU: N/A
- Others: N/A
```

### Current Behavior

When milvus starts, channels are assigned to data nodes.

Datacoord has Etcd store channel watch information as k-v pairs, where key = nodeID + channelName, value = an individual channel watch info:
https://github.com/milvus-io/milvus/blob/09b2c46b619ef554a538a508569b7e6f52503e0b/internal/datacoord/channel_store.go#L324-L345

An individual channel watch info has the structure of:
message ChannelWatchInfo {
    VchannelInfo vchan= 1;
    int64 startTs = 2;
    ChannelWatchState state = 3;
    int64 timeoutTs = 4;
}

furthermore, VchannelInfo has the structure of:

message VchannelInfo {
  int64 collectionID = 1;
  string channelName = 2;
  internal.MsgPosition seek_position = 3;
  repeated SegmentInfo unflushedSegments = 4;
  repeated SegmentInfo flushedSegments = 5;
  repeated SegmentInfo dropped_segments = 6;
}

which contains information of all unflashed, flashed and dropping segments (so that is..basically every single segment that has anything to do with this vchannel IMO?) information , each of which looks like:

message SegmentInfo {
  int64 ID = 1;
  int64 collectionID = 2;
  int64 partitionID = 3;
  string insert_channel = 4;
  int64 num_of_rows = 5;
  common.SegmentState state = 6;
  int64 max_row_num = 7;
  uint64 last_expire_time = 8;
  internal.MsgPosition start_position = 9;
  internal.MsgPosition dml_position = 10;
  repeated FieldBinlog binlogs = 11;
  repeated FieldBinlog statslogs = 12;
  repeated FieldBinlog deltalogs = 13;
  bool createdByCompaction = 14;
  repeated int64 compactionFrom = 15;
  uint64 dropped_at = 16;
}

All these segment info, adding up, would make a single etcd k-v pair quite huge.

When Milvus restarts, for example, new data nodes are brought up. This makes datacoord to reload, reassign and store node-channels mapping (the huge k-v pairs) from/into Etcd, which all of a sudden could cause a huge amount of Etcd traffic.

### Expected Behavior

Not sure if there are any other cases causing other types of huge k-v pair in Etcd. But at least this one is.

IMO channel watch info should be restructured and irrelevant information should not be store in channel watch info.

After all, as mentioned on their official website: "Etcd is designed to handle small key value pairs typical for metadata."

**Comment by xiaofan-luan** [comment]
@sunby @congqixia 
My suggestions is to move SegmentInfo out of vchannelinfo but only leave Segment ID.
We maintain another kv pair to store the relation between segment and SegmentInfo, any advices?

**Comment by congqixia** [comment]
@xiaofan-luan @soothing-rain we might trim the segment path part, but the position information is needed in datanode.

**Comment by xiaofan-luan** [comment]
My suggestion is to separate KV
VchannelInfo -> {
int64 collectionID = 1;
string channelName = 2;
internal.MsgPosition seek_position = 3;
repeated SegmentId unflushedSegments = 4;
repeated SegmentId flushedSegments = 5;
repeated SegmentId dropped_segments = 6;
}

and segment info is stored in another KV.
When write we can to txn if need to modify both kvs.

On Read we can do a join in memory

**Comment by soothing-rain** [comment]
> My suggestion is to separate KV VchannelInfo -> { int64 collectionID = 1; string channelName = 2; internal.MsgPosition seek_position = 3; repeated SegmentId unflushedSegments = 4; repeated SegmentId flushedSegments = 5; repeated SegmentId dropped_segments = 6; }
> 
> and segment info is stored in another KV. When write we can to txn if need to modify both kvs.
> 
> On Read we can do a join in memory

Yes, I don't think we need the whole VchannelInfo for a `channelwatch` store.

@congqixia How about we just replace VchannelInfo in `ChannelWatchInfo` with a simplified version including only collectionID and channelName, while keeping other VchannelInfo (used in datanode, flowgraph, etc.) untouched. How does that sound?

**Comment by xiaofan-luan** [comment]
@soothing-rain any progress on this one?

**Comment by soothing-rain** [comment]
> @soothing-rain any progress on this one?

Upon our last discussion, we have deprioritized this task and the new plan is to take a deep look at it and fix it in v2.1.x.

**Comment by xiaofan-luan** [comment]
trim seems to actually work on most cases, but compaction has to ensure segment number is in control on that case

**Comment by xiaofan-luan** [comment]
should be fixed by the compaction policy and #17441,  close for now
